jsk_pcl_ros/CMakeLists.txt

```diff
@@ -7,21 +7,46 @@ project(jsk_pcl_ros)
 set(PCL_MSGS pcl_msgs) ## hydro and later
 
 find_package(catkin REQUIRED COMPONENTS
-  jsk_recognition_utils jsk_pcl_ros_utils
-  dynamic_reconfigure pcl_ros nodelet message_generation genmsg
-  ${PCL_MSGS} sensor_msgs geometry_msgs jsk_recognition_msgs
-  eigen_conversions tf_conversions tf2_ros tf
-  image_transport nodelet cv_bridge
-  jsk_topic_tools
+  cv_bridge
+  diagnostic_msgs
+  diagnostic_updater
+  dynamic_reconfigure
+  eigen_conversions
+  genmsg
+  geometry_msgs
   image_geometry
-  jsk_footstep_msgs
+  image_transport
+  image_view2
   interactive_markers
+  jsk_footstep_msgs
+  jsk_pcl_ros_utils
+  jsk_recognition_msgs
+  jsk_recognition_utils
+  jsk_topic_tools
+  kdl_conversions
+  kdl_parser
   laser_assembler
-  octomap_server
-  octomap_ros
+  message_generation
+  ml_classifiers
+  moveit_core
+  nav_msgs
+  nodelet
   octomap_msgs
-  kdl_parser
-  kdl_conversions
+  octomap_ros
+  octomap_server
+  pcl_conversions
+  ${PCL_MSGS}
+  pcl_ros nodelet
+  rosboost_cfg
+  roscpp_tutorials
+  sensor_msgs
+  std_msgs
```

Comment:
i don't think explicit depends does not requires `find_package` nor
`catkin_package`
for example
https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/velocity_controllers/include/velocity_controllers/joint_position_controller.h#L78
uses `#include <std_msgs/Float64.h` but `std_msgs` is not included in
https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/velocity_controllers/CMakeLists.txt#L5-L27

--
◉ Kei Okada


On Sun, Aug 7, 2016 at 11:08 PM, Kentaro Wada <notifications@github.com>
wrote:

> In jsk_pcl_ros/CMakeLists.txt
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1828#discussion_r73802595>
> :
>
> >    octomap_msgs
> > -  kdl_parser
> > -  kdl_conversions
> > +  octomap_ros
> > +  octomap_server
> > +  pcl_conversions
> > +  ${PCL_MSGS}
> > +  pcl_ros nodelet
> > +  rosboost_cfg
> > +  roscpp_tutorials
> > +  sensor_msgs
> > +  std_msgs
>
> I seems that jsk_pcl_ros explicitly depends on std_msgs, and this is
> necessary.
>
> CMakeLists.txt:  std_msgs
> CMakeLists.txt:  DEPENDENCIES std_msgs sensor_msgs geometry_msgs jsk_recognition_msgs
> CMakeLists.txt:  CATKIN_DEPENDS std_msgs sensor_msgs geometry_msgs message_runtime jsk_recognition_msgs jsk_recognition_utils
> euslisp//detection_interface.l:                                 (instance std_msgs::header :init
> euslisp//detection_interface.l:                                 (instance std_msgs::header :init
> euslisp//detection_interface.l:                                 (instance std_msgs::header :init
> euslisp//kalman-filtered-objectdetection-marker.l:        (header (instance std_msgs::header :init))
> euslisp//kalman-filtered-objectdetection-marker.l:(ros::subscribe "input/switch" std_msgs::Bool #'detect-switch-cb)
> include//jsk_perception/color_histogram.h:                            const std_msgs::Header& header);
> include//jsk_perception/color_histogram.h:                            const std_msgs::Header& header);
> include//jsk_perception/color_histogram.h:                            const std_msgs::Header& header);
> include//jsk_perception/color_histogram.h:                            const std_msgs::Header& header);
> include//jsk_perception/grid_label.h:    virtual void makeLabel(cv::Mat& label, const std_msgs::Header& header);
> node_scripts//image_saver_sync:from std_msgs.msg import Header
> node_scripts//image_time_diff.py:from std_msgs.msg import Header
> node_scripts//publish_header:"""Publish std_msgs/Header with current timestamp.
> node_scripts//publish_header:from std_msgs.msg import Header
> node_scripts//rect_array_to_image_marker.py:from std_msgs.msg import ColorRGBA
> package.xml:  <build_depend>std_msgs</build_depend>
> package.xml:  <run_depend>std_msgs</run_depend>
> Binary file sample//data/2016-06-10-23-01-28_in_lab.bag matches
> Binary file sample//data/2016-06-10-23-01-28_in_lab.orig.bag matches
> src//color_histogram.cpp:                                  const std_msgs::Header& header)
> src//color_histogram.cpp:                                  const std_msgs::Header& header)
> src//color_histogram.cpp:                                  const std_msgs::Header& header)
> src//color_histogram.cpp:                                  const std_msgs::Header& header)
> src//color_histogram_label_match.cpp:      std_msgs::ColorRGBA coef_color = jsk_topic_tools::heatColor(coef);
> src//colorize_float_image.cpp:          std_msgs::ColorRGBA c = jsk_topic_tools::heatColor((v - min_value) / (max_value - min_value));
> src//grid_label.cpp:  void GridLabel::makeLabel(cv::Mat& label, const std_msgs::Header& header) {
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1828/files/5c8d7d04c921a71dee29025a37f9da0e73143e4e..26cb133674dfb49b96f46dda7e38e77d67530d87#r73802595>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3PD0KHBg3Vik3B_Ez9wAQtuf2RBIks5qdebzgaJpZM4JeUkQ>
> .
>